src/utstring.h

```diff
@@ -117,7 +117,7 @@ do {                                                             \
   (dst)->d[(dst)->i]='\0';                                       \
 } while(0)
 
-#define utstring_len(s) ((unsigned)((s)->i))
+#define utstring_len(s) ((size_t)((s)->i))
```

Comment:
Oho, now I know what'll break if we remove it.

(1) Many test cases call `printf("%u", utstring_len(s))`; this needs to change to `printf("%zu", utstring_len(s));`, and then we lose C89 compatibility because `%zu` is a C99 invention. The proper workaround there is probably to add back the cast: `printf("%u", (unsigned)utstring_len(s))`. We need to do *something* to fix the warnings from the test suite, anyway.

(2) In C++, the type of an expression affects overload resolution; e.g. `std::max(100u, utstring_len(s))` will no longer compile. However, IMHO nobody should even *think* about using utstring in C++ given that `std::string` is available! So I'm willing to punt on this one.

I'll go ahead and merge this patch with the above fix to issue (1), unless someone objects first.